server/src/main/kotlin/fi/oph/kitu/yki/responses/YkiSuoritusResponse.kt

```diff
@@ -19,31 +24,21 @@ import com.fasterxml.jackson.annotation.JsonPropertyOrder
     "puhuminen",
     "yleisarvosana",
 )
-data class YkiSuoritus(
+@Features(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
+class YkiSuoritusResponse(
     @JsonProperty("suorittajanOppijanumero")
```

Comment:
Kommentoin nyt vaikka nää ei ole tästä pullarista suoraan 😅 Mutta tarviiko näitä annotaatioita, kun kun toi propertyn nimi on sama sekä csv:ssä että koodissa?

Reply:
Jackson olettaa, että responsessa enumit on kirjoitettu caps lockilla.
https://stackoverflow.com/questions/24157817/jackson-databind-enum-case-insensitive

 Toinen vaihtoehto olis varmaan kirjoittaa enum - luokassa arvot pienellä, mut siitä taitaa tulle linteri virheitä. Kokeilin myös

```kotlin
enum class MyEnum {
    Foo("foo"),
    Bar("bar"),
    ;
}
```

eikä siitäkään ollut apua. Päädyin sitten muokkaamaan sitä csv-parsemista, mutta en halunnut kovakoodata sitä parseriin jos joku toinen rajapinta halutaan konffia eri tavalla. Kirjoitin oman annotaation, jotta featuret voi olla rajapinta - kohtaisia.

Reply:
Aa sori olin vähän epäselvä, meinasin näitä `@JsonProperty` annotaatioita

Reply:
Tuo jackson - kirjasto vaatii ne. Testasin vielä kertaalleen ja ilman niitä tulee
```
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Invalid type definition for type `fi.oph.kitu.yki.responses.YkiSuoritusResponse`: Argument #0 of Creator [constructor for `fi.oph.kitu.yki.responses.YkiSuoritusResponse` (14 args), annotations: [null] has no property name (and is not Injectable): can not use as property-based Creator
 at [Source: (StringReader); line: UNKNOWN, column: UNKNOWN]
```

Reply:
Vaikuttaisiko tähän muuten että jos tuo olisi `data class`?

Reply:
Tää on siis siitä jännä että JSONin purku kyllä toimii ilman näitä annotaatioita. 😅

Reply:
@wolverian testattu, ja `data class`:lla ei ole merkitystä.